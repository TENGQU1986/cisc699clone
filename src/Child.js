import React, {useEffect, memo, useRef} from 'react'
import {isEqual} from 'lodash'

const Child = memo((props) => {
    const {counter, list} = props
    const childRef = useRef(0)
    useEffect(() => {
        console.log('Child component rendered!')
        // childRef.current = childRef.current + 1
    })
    return (
        <div className="Child">
            <h1>Child with Memoization - {counter}</h1>
            
        </div>
    )
}, isEqual)

export default Child
