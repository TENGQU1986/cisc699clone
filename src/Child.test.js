import React from 'react';
import Child from './Child';
import { render, fireEvent, act } from "@testing-library/react";


it('should render Child component', () => {
    const { container } = render(<Child />);
    expect(container).toMatchSnapshot();
})

// describe('clickButton', () => {
//     it("on click", () => {
//         const { queryByTitle } = render(<Child />);
//         const btn = queryByTitle("parent_button");
//         expect(btn.innerHTML).toBe("switch to memo")
//         fireEvent.click(btn);
//         expect(btn.innerHTML).toBe("switch to unmemo")
        
//     })

//     it('on input change', async () => {
//         const onInputChange = jest.fn();
//         const { queryByTitle } = render(<Child />);
//         await act(async () => {
//             fireEvent.input(await queryByTitle("parent_input"))
//         })
//         const input = await queryByTitle("parent_input");
        
//         expect(Child).toMatchSnapshot();
//     })
// })