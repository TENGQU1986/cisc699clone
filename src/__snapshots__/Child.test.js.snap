// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Child component 1`] = `
<div>
  <div
    class="Child"
  >
    <h1>
      Child with Memoization - 
    </h1>
  </div>
</div>
`;
